File: Data Structures & Algorithms/maximum-product-subarray/submission-2.java
class Solution {
    public int maxProduct(int[] nums) {
        int max=nums[0];
        int min=nums[0];
        int res=nums[0];
        int n=nums.length;
        for(int i=1;i<n;i++){
            int oldmax=max;
            int oldmin=min;
            max=Math.max(nums[i],Math.max(nums[i]*oldmax,nums[i]*oldmin));
            min=Math.min(nums[i],Math.min(nums[i]*oldmax,nums[i]*oldmin));

            res=Math.max(res,max);
        }
        return res;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i=0;i<nums.size();i++){
            int l=target-nums[i]; 
            for(int j=i+1;j<nums.size();j++){  
                if(l==nums[j]){
                   vector <int>a;
                    a.push_back(i);
                    a.push_back(j);
                    return a;
                }
            }
        }
    }

};
